### PR TITLE
ramips: add support for Keenetic kn-1210

### PR DESCRIPTION
**Pull request by frol1805**

### CODE REVIEW

**Comment by adrianschmutzler** [comment]
This pull request has formal issues. Please have another look at
https://openwrt.org/submitting-patches

File: target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds [hunk 1]
```diff
@@ -135,6 +135,10 @@ zyxel,keenetic-extra-ii)
 	set_wifi_led "$boardname:green:wifi"
 	ucidef_set_led_switch "internet" "internet" "$boardname:green:internet" "switch0" "0x01"
 	;;
+zyxel,keenetic-kn-1210)
+	set_wifi_led "$boardname:green:wifi"
```
Comment (adrianschmutzler) [inline target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds:139]: Please use trigger in DTS instead.

File: target/linux/ramips/mt76x8/base-files/etc/board.d/02_network [hunk 2]
```diff
@@ -152,6 +152,10 @@ ramips_setup_macs()
 	zyxel,keenetic-extra-ii)
 		wan_mac=$(macaddr_add "$(mtd_get_mac_binary factory 0x4)" 1)
 		;;
+	zyxel,keenetic-kn-1210)
+		ucidef_add_switch "switch0" \
+			"1:lan" "2:lan" "3:lan" "0:wan" "6@eth0"
+		;;
```
Comment (adrianschmutzler) [inline target/linux/ramips/mt76x8/base-files/etc/board.d/02_network:158]: This section is for mac addresses. Move this to the proper section.

File: target/linux/ramips/image/mt76x8.mk [hunk 3]
```diff
@@ -618,3 +618,15 @@ define Device/zyxel_keenetic-extra-ii
 	check-size | zyimage -d 6162 -v "ZyXEL Keenetic Extra II"
 endef
 TARGET_DEVICES += zyxel_keenetic-extra-ii
+
+define Device/zyxel_keenetic-kn-1210
+  DTS := kn-1210
```
Comment (adrianschmutzler) [inline target/linux/ramips/image/mt76x8.mk:623]: This line can be dropped.

File: target/linux/ramips/image/mt76x8.mk [hunk 4]
```diff
@@ -618,3 +618,15 @@ define Device/zyxel_keenetic-extra-ii
...
+  IMAGE_SIZE := 16064k
+  BLOCKSIZE := 64k
+  DEVICE_TITLE := Keenetic 4G (KN-1210)
```
Comment (adrianschmutzler) [inline target/linux/ramips/image/mt76x8.mk:626]: Use DEVICE_VENDOR/DEVICE_MODEL instead.

File: target/linux/ramips/image/mt76x8.mk [hunk 5]
```diff
@@ -618,3 +618,15 @@ define Device/zyxel_keenetic-extra-ii
...
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-usb-ledtrig-usbport
+  IMAGES += factory.bin
+  IMAGE/factory.bin := $$(sysupgrade_bin) | pad-to $$$$(BLOCKSIZE) | \
+	check-size $$$$(IMAGE_SIZE) | zyimage -d 4624 -v "KN-1210"
```
Comment (adrianschmutzler) [inline target/linux/ramips/image/mt76x8.mk:630]: check-size works without argument.

File: target/linux/ramips/image/mt76x8.mk (hunk 4 above)
Comment (adrianschmutzler) [reply target/linux/ramips/image/mt76x8.mk:626]: Is the "Keenetic 4G" name unique? I'd rather this one for the image name, as the other one is only given in brackets when looking for this device...

File: target/linux/ramips/dts/mt7628an_zyxel_keenetic-kn-1210.dts [hunk 6]
```diff
@@ -0,0 +1,599 @@
+/dts-v1/;
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_zyxel_keenetic-kn-1210.dts:1]: please add an SPDX license identifier.

**Comment by adrianschmutzler** [comment]
Please have a look at other DTS files and how they are used typically. Primarily, you should include mt76x8.dtsi and inherit most of the content from it. Despite, try to look at the style of the other DTS files and design yours based on what you find there.

File: target/linux/ramips/image/mt76x8.mk (hunk 4 above)
Comment (frol1805) [reply target/linux/ramips/image/mt76x8.mk]: fix

File: target/linux/ramips/image/mt76x8.mk (hunk 5 above)
Comment (frol1805) [reply target/linux/ramips/image/mt76x8.mk]: fix

File: target/linux/ramips/dts/mt7628an_zyxel_keenetic-kn-1210.dts (hunk 6 above)
Comment (frol1805) [reply target/linux/ramips/dts/mt7628an_zyxel_keenetic-kn-1210.dts]: fix

File: target/linux/ramips/image/mt76x8.mk (hunk 3 above)
Comment (frol1805) [reply target/linux/ramips/image/mt76x8.mk]: fix

File: target/linux/ramips/mt76x8/base-files/etc/board.d/02_network (hunk 2 above)
Comment (frol1805) [reply target/linux/ramips/mt76x8/base-files/etc/board.d/02_network]: fix

File: target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds (hunk 1 above)
Comment (frol1805) [reply target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds:139]: fix

**Pull request closed without merging**